proposals/string-ascii-unicode-utf-grapheme.md

```diff
@@ -0,0 +1,189 @@
+
+## UTF standards
+
+- UTF-8 is 1-4 sets of 8 bits long
+- UTF-16 is 1-2 sets of 16 bits long
+- UTF-32 is 1 set of 32 bits long
+
+### When slicing by unicode codepoint e.g. "🔥🔥🔥" (\U0001f525\U0001f525\U0001f525)
+
+- UTF-8: 12 sets (12 bytes) long. The first byte of each fire can be used to
+know the length, and the next bytes are what is known as a continuation byte.
+There are several approaches to achieve the slicing, they can be explored with
+benchmarking later on.
+- UTF-16: 6 sets long. It's very similar in procedure to UTF-8.
+- UTF-32: It is fastest since it's direct index access. This is not the case
+when supporting graphemes.
+
+## Graphemes
+
+Graphemes are an extension which allows unicode codepoints to modify the end
+result through concatenation with other codepoints. Visually they are shown as a
+single unit e.g. é is actually comprised of `e` followed by `´`  (\x65\u0301).
+
+Graphemes are more expensive to slice because one can't use faster algorithms,
+only skipping each unicode codepoint one at a time and checking the last byte
+to see if it's an extended cluster *.
+
+*: extended clusters is the full feature complete grapheme standard. There
+exists the posibility of implementing only a subset, but that defeats the
+purpose of going through the hassle of supporting it in the first place.
+
+## Context
+
+C, C++, and Rust use UTF-8 with no default support for graphemes.
+
+### Swift
+
+Swift is an interesting case study. For compatibility with Objective-C they went
+for UTF-16. They decided to support grapheme clusters by default. They recently
+changed the preferred encoding from UTF-16 to UTF-8. They have a `Character`
+that I think inspired our current `Char` type, it is a generic representation of
+a Character in any encoding, that can be from one codepoint up to any grapheme
+cluster.
+
+### Python
+
+Python currently uses UTF-32 for its string type. So the slicing and indexing
+is simple and fast (but consumes a lot of memory, they have some tricks to
+reduce it). They do not support graphemes by default. Pypi is implementing a
+UTF-8 version of Python strings, which keeps the length state every x
+characters.
+
+### Mojo
+
+Mojo aims to be close to Python yet be faster and customizable, taking advantage
+of heterogeneous hardware and modern type system features.
+
+## Value vs. Reference
+
+Our current `Char` type uses a u32 as storage, every time an iterator that
+yields `Char`is used, an instance is parsed from the internal UTF-8 encoded
+`StringSlice` (into UTF-32).
+
+The default iterator for `String` returns a `StringSlice` which is a view into
+the character in the UTF-8 encoded `StringSlice`. This is much more efficient
+and does not add any complexity into the type system nor developer headspace.
+
+## Now, onto the Proposal
+
+### Goals
+
+#### Hold off on developing Char further and remove it from stdlib.builtin
+
+`Char` is currently expensive to create and use compared to a `StringSlice`
```

Comment:
Do we have numbers on this? The creation but also using different methods on it. Let's avoid optimizing things without measuring all the workflows first.